### ISSUE TITLE
[Bug]: Build errors on iOS with XCode 14.3

### ISSUE THREAD

**Issue by indiecore** [braze-inc/braze-unity-sdk#101]
### Braze Unity SDK Version

4.1.1

### Steps To Reproduce

* Import 4.1.1 into project.
* Build iOS
* Try to build Unity project with Xcode 14.3

### Expected Behavior

Expected the project to build correctly

### Actual Incorrect Behavior

Project did not build correctly.

![Screenshot 2023-06-27 at 10 45 04 AM](https://github.com/braze-inc/braze-unity-sdk/assets/1449818/f871abcf-065d-4a6d-8e18-6b4431ea61f7)

### Verbose Logs

```shell
[Project]/Libraries/external/baselib/Include/Cpp/Internal/ConditionVariable_SemaphoreBased.inl.h
[Project]/Libraries/external/baselib/Include/Cpp/Internal/ConditionVariable_SemaphoreBased.inl.h:15:56 Unknown type name 'timeout_ms'

[Project]/Libraries/external/baselib/Include/Cpp/ConditionVariable.h
[Project]/Libraries/external/baselib/Include/Cpp/ConditionVariable.h:68:41 Unknown type name 'timeout_ms'

[Project]/Libraries/external/baselib/Include/Cpp/Lock.h
[Project]/Libraries/external/baselib/Include/Cpp/Lock.h:74:53 Unknown type name 'timeout_ms'

[Project]/Libraries/external/baselib/Include/Cpp/Lock.h:148:59 Unknown type name 'timeout_ms'

[Project]/Libraries/external/baselib/Include/Cpp/ReentrantLock.h
[Project]/Libraries/external/baselib/Include/Cpp/ReentrantLock.h:87:53 Unknown type name 'timeout_ms'

[Project]/Libraries/external/baselib/Include/Cpp/ReentrantLock.h:162:59 Unknown type name 'timeout_ms'

[Project]/Libraries/external/baselib/Include/Cpp/Semaphore.h
[Project]/Libraries/external/baselib/Include/Cpp/Semaphore.h:85:47 Unknown type name 'timeout_ms'

[Project]/Libraries/external/baselib/Include/Cpp/Time.h
[Project]/Libraries/external/baselib/Include/Cpp/Time.h:11:65 No member named 'milli' in namespace 'std'

[Project]/Libraries/external/baselib/Include/Cpp/Time.h:12:65 No member named 'micro' in namespace 'std'

[Project]/Libraries/external/baselib/Include/Cpp/Time.h:16:65 No member named 'nano' in namespace 'std'

[Project]/Libraries/external/baselib/Include/Cpp/Time.h:17:78 Use of undeclared identifier 'duration'; did you mean 'std::chrono::duration'?

[Project]/Libraries/external/baselib/Include/Cpp/Time.h:17:78 Use of undeclared identifier 'duration'

[Project]/Libraries/external/baselib/Include/Cpp/Time.h:18:25 Use of undeclared identifier 'duration'

[Project]/Libraries/external/baselib/Include/Cpp/Time.h:19:28 Use of undeclared identifier 'duration'

[Project]/Libraries/external/baselib/Include/Cpp/Time.h:23:20 Unknown type name 'time_point'

[Project]/Libraries/external/baselib/Include/Cpp/Time.h:33:20 Unknown type name 'duration'

[Project]/Libraries/external/baselib/Include/Cpp/Time.h:35:24 Use of undeclared identifier 'duration'

[Project]/Libraries/external/baselib/Include/Cpp/Time.h:38:68 Unknown type name 'duration'

[Project]/Libraries/external/baselib/Include/Cpp/Time.h:44:20 Unknown type name 'time_point'
```

### Additional Information

The only difference between builds was the upgrade to Braze 4.1.1 from Braze 3.11.0.

**Comment by hokstuff** [comment]
Hi @indiecore,

Thanks for bringing this to our attention. We have been able to reproduce this issue when building on Xcode 14.3 and suspect it may have to do with some of the compiler changes mentioned in [that version's release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-14_3-release-notes). We are investigating this issue and will update this thread when we find out more information. In the meantime, feel free to continue using Xcode 14.2 as a workaround, as compilation should still work on that version.

**Comment by plattnum** [comment]
Its been almost 5 months since this has been reported. Is there any update to this. xCode and MacOS versions are marching along at a steady pace. I am in a situation where I won't be able to downgrade my xCode or MacOS. I meet this issue on xCode 15 and Sonoma.

**Comment by roryos-drest** [comment]
@hokstuff any update on this?
as raised its been 5 months.
Apart from downgrading which is not a long term solution what are the alternatives given its been mentioned this is an issue on XCode 15 as well making the whole of Braze unusable going forward?

**Comment by hokstuff** [comment]
Hi @plattnum @roryos-drest @indiecore,

Thanks for your patience and for raising this again. We actually are in the works of in production-izing additional release assets in the [native Swift SDK](https://github.com/braze-inc/braze-swift-sdk/) to support new integration options. This has also been tested locally and appears to resolve the root cause of the build issues for this thread as well as [this one](https://github.com/braze-inc/braze-unity-sdk/issues/102). We will continue to validate this change on our end and will be in touch once we are able to bring the change over to the Unity SDK.

**Comment by hokstuff** [comment]
Hi @indiecore @plattnum @roryos-drest,

We have released [Unity SDK version 5.0.0](https://github.com/braze-inc/braze-unity-sdk/releases/tag/5.0.0) which addresses the issues with Xcode 14.3+ (it has also been verified with Xcode 15.x). Let us know if you run into any issues!

Thanks!
